src/test/java/bio/terra/service/snapshotbuilder/query/QueryTest.java

```diff
@@ -122,4 +129,82 @@ void renderComplexSQL() {
                 "(c.condition_concept_id = 316139 OR c0.ancestor_concept_id = 316139 OR c.condition_concept_id = 4311280 OR c0.ancestor_concept_id = 4311280)"),
             containsString("AND p.year_of_birth < 1983")));
   }
+
+  @Test
+  void renderSQLWithDatasetModelGCP() {
+    DatasetModel dataset =
+        new DatasetModel().name("name").dataProject("project").cloudPlatform(CloudPlatform.GCP);
+    Query query = buildGetConceptsQuery(100, generateTableNameGCP(dataset));
+    String sql = query.renderSQL();
+    assertThat(
+        sql,
+        is(
+            "SELECT c.concept_name, c.concept_id FROM `project.datarepo_name.concept` AS c "
+                + "WHERE c.concept_id IN "
+                + "(SELECT c.descendant_concept_id FROM `project.datarepo_name.concept_ancestor` AS c "
+                + "WHERE c.ancestor_concept_id = 100)"));
+  }
+
+  @Test
+  void renderSQLWithDatasetModelAzure() {
+    String datasetSource = "source_dataset_data_source_0";
+    Query query = buildGetConceptsQuery(100, generateTableNameAzure(datasetSource));
+    String sql = query.renderSQL();
+    assertThat(
+        sql,
+        is(
+            "SELECT c.concept_name, c.concept_id FROM "
+                + "(SELECT * FROM OPENROWSET(BULK 'metadata/parquet/concept/*/*.parquet', DATA_SOURCE = 'source_dataset_data_source_0', FORMAT = 'parquet') AS alias951024263) "
+                + "AS c WHERE c.concept_id IN "
+                + "(SELECT c.descendant_concept_id FROM "
+                + "(SELECT * FROM OPENROWSET(BULK 'metadata/parquet/concept_ancestor/*/*.parquet', DATA_SOURCE = 'source_dataset_data_source_0', FORMAT = 'parquet') AS alias625571305) "
+                + "AS c WHERE c.ancestor_concept_id = 100)"));
+  }
+
+  public static Function<String, String> generateTableNameGCP(DatasetModel dataset) {
+    return (tableName) -> {
+      String dataProjectId = dataset.getDataProject();
+      String bqDatasetName = PdaoConstant.PDAO_PREFIX + dataset.getName();
+      return String.format("`%s.%s.%s`", dataProjectId, bqDatasetName, tableName);
+    };
+  }
+
+  public static Function<String, String> generateTableNameAzure(String sourceDatasetDatasource) {
+    return (tableName) -> {
+      String alias = "alias" + Math.abs(Objects.hash(tableName));
```

Comment:
Is this inner `as` needed? The query you generate is
```
SELECT xx FROM (SELECT * from OPENROWSET(...) AS alias951024263) AS c WHERE c.concept_id IN...
```
So `alias951024263` isn't used in the SQL expression.